### PR TITLE
Threshold flow/v8

### PR DESCRIPTION
**Pull request by victorjulien**
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1822

https://redmine.openinfosecfoundation.org/issues/6822

replacing #10910, rebasing and addressing a comment

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20525

**Review by catenacyber — APPROVED** [review]
Thanks for the work :-)

- CI : 🟠 probably just need a rebase
- Code : Checking now
- Commits segmentation : nice, maybe the commit `detect/content: fix wrong value for depth check` should come before the commits reordering keyword id's
- Commit messages : nice
- Git ID set : looks fine for me
- CLA : you already contributed :-p
- Doc update : ok
- Redmine ticket : ok
- Rustfmt : not needed
- Tests : left some remarks there about readability, but fair enough for proving the suricata PR works
- Dependencies added: none

File: src/util-var.h [hunk 1]
```diff
@@ -46,7 +47,7 @@ enum VarTypes {
 };
 
 typedef struct GenericVar_ {
-    uint8_t type;
+    uint8_t type; /**< variable type, uses detection sm_type */
```
Comment (catenacyber) [inline src/util-var.h:51]: Should we not precise that `sm_type` is u16 but the order of keywords makes it sure that we fit into an u8 ?

**Comment by catenacyber** [comment]
Should you rebase this to get green CI ?

**Comment by victorjulien** [comment]
replaced by #11254

**Pull request closed without merging**